components/dashboard/src/components/DropDown2.tsx

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import React, { FunctionComponent, RefObject, useCallback, useEffect, useMemo, useRef, useState } from "react";
+import Arrow from "./Arrow";
+
+export interface DropDown2Element {
+    id: string;
+    element: JSX.Element;
+    isSelectable?: boolean;
+}
+
+export interface DropDown2Props {
+    getElements: (searchString: string) => DropDown2Element[];
+    searchPlaceholder?: string;
+    disableSearch?: boolean;
+    expanded?: boolean;
+    onSelectionChange: (id: string) => void;
+}
+
+export const DropDown2: FunctionComponent<DropDown2Props> = (props) => {
+    const [showDropDown, setShowDropDown] = useState<boolean>(!!props.expanded);
+    const nodeRef: RefObject<HTMLDivElement> = useRef(null);
+    const onSelected = useCallback(
+        (elementId: string) => {
+            props.onSelectionChange(elementId);
+            setShowDropDown(false);
+        },
+        [props],
+    );
+    const [search, setSearch] = useState<string>("");
+    const filteredOptions = useMemo(() => props.getElements(search), [props, search]);
+    const [selectedElementTemp, setSelectedElementTemp] = useState<string | undefined>(filteredOptions[0]?.id);
+
+    // reset search when the drop down is expanded or closed
+    useEffect(() => {
+        setSearch("");
+        if (showDropDown && selectedElementTemp) {
+            document.getElementById(selectedElementTemp)?.scrollIntoView({ behavior: "smooth", block: "nearest" });
+        }
+        // we only want this behavior when showDropDown changes to true.
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [showDropDown]);
+
+    const toggleDropDown = useCallback(() => {
+        setShowDropDown(!showDropDown);
+    }, [setShowDropDown, showDropDown]);
+
+    const setFocussedElement = useCallback(
+        (element: string) => {
+            setSelectedElementTemp(element);
+            document.getElementById(element)?.scrollIntoView({ behavior: "smooth", block: "nearest" });
+            document.getElementById(element)?.focus();
+        },
+        [setSelectedElementTemp],
+    );
+
+    const onKeyDown = useCallback(
+        (e: React.KeyboardEvent) => {
+            if (showDropDown && e.key === "ArrowDown") {
+                e.preventDefault();
+                let idx = filteredOptions.findIndex((e) => e.id === selectedElementTemp);
+                while (idx++ < filteredOptions.length - 1) {
```

Comment:
It took me awhile to understand why we needed a loop here. I think it's so that we select the next "selectable" option, since not all options are selectable. If I'm understanding that correctly, maybe a comment to help future-us understand it a bit would be helpful?